commands/remove.rb

```diff
@@ -34,17 +45,42 @@ def self.remove(pkg, verbose)
     # Remove the files and directories installed by the package.
     unless pkg.is_fake?
       Dir.chdir CREW_CONFIG_PATH do
-        # Remove all files installed by the package.
+        # Remove all files installed by the package in CREW_PREFIX and
+        # HOME.
+        # Exceptions:
+        # 1. The file exists in another installed package.
+        # 2. The file is in one of the filelists for packages in
+        # CREW_ESSENTIAL_FILES, or a dependendent package of
+        # CREW_ESSENTIAL_PACKAGES.
         flist = File.join(CREW_META_PATH, "#{pkg.name}.filelist")
         if File.file?(flist)
-          File.foreach(flist, chomp: true) do |line|
-            next unless line.start_with?(CREW_PREFIX)
-            if system("grep --exclude #{pkg.name}.filelist -Fxq '#{line}' ./meta/*.filelist")
-              puts "#{line} is in another package. It will not be removed during the removal of #{pkg.name}.".orange
-            else
-              puts "Removing file #{line}".yellow if verbose
-              FileUtils.remove_file line, exception: false
-            end
+          # When searching for files to delete we exclude the files from
+          # all packages and dependent packages of CREW_ESSENTIAL_PACKAGES.
+          essential_deps_exclude_froms = essential_deps.map { |i| File.file?("#{File.join(CREW_META_PATH, i.to_s)}.filelist") ? "--exclude-from=#{File.join(CREW_META_PATH, i.to_s)}.filelist" : '' }.join(' ')
+
+          # When making a list of all files from crew filelists we again
+          # ignore all files from packages and dependent packages of
+          # CREW_ESSENTIAL_PACKAGES.
+          essential_deps_excludes = essential_deps.map { |i| File.file?("#{File.join(CREW_META_PATH, i.to_s)}.filelist") ? "--exclude=#{File.join(CREW_META_PATH, i.to_s)}.filelist" : '' }.join(' ')
+
+          package_files_cmd = "grep -h #{essential_deps_exclude_froms} \"^#{CREW_PREFIX}\\|^#{HOME}\" #{flist}"
+          package_files = `#{package_files_cmd}`.split("\n").uniq.sort
```

Comment:
Is there any reason to have the `package_files_cmd` variable here instead of just doing it in one line?
i.e.
```
package_files = `#{"grep -h #{essential_deps_exclude_froms} \"^#{CREW_PREFIX}\\|^#{HOME}\" #{flist}"}`.split("\n").uniq.sort
```

Reply:
It can be in the same line now that it is simplified!